apis/common/parent/projectlocation.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package parent
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ Parent = &ProjectAndLocationParent{}
+
+type ProjectAndLocationParent struct {
```

Comment:
We can merge `ProjectAndLocationParent` and `ProjectAndLocationRef`, it makes this file shorter. But it adds additional complexity to the controller on error handling,  and the order of the function calls to set the `external`.

Reply:
I think we want Foo, FooRef and FooIdentity.  I think this is FooIdentity, and ProjectAndLocationRef is FooRef